lib/signet/typed.ex

```diff
@@ -126,16 +130,33 @@ defmodule Signet.Typed do
         iex> Signet.Typed.Type.deserialize_type("bytes32")
         {:bytes, 32}
 
+        iex> Signet.Typed.Type.deserialize_type("bytes32[]")
+        {:list, {:bytes, 32}}
+
         iex> Signet.Typed.Type.deserialize_type("Person")
         "Person"
+
+        iex> Signet.Typed.Type.deserialize_type("bag")
+        ** (RuntimeError) unknown type: bag
     """
     @spec deserialize_type(String.t()) :: field_type()
     def deserialize_type("address"), do: :address
     def deserialize_type("uint256"), do: {:uint, 256}
     def deserialize_type("bytes32"), do: {:bytes, 32}
     def deserialize_type("string"), do: :string
     def deserialize_type("bytes"), do: :bytes
-    def deserialize_type(custom_type) when is_binary(custom_type), do: custom_type
+    def deserialize_type(ty) when is_binary(ty) do
+      cond do
+        String.ends_with?(ty, "[]") ->
+          {:list, deserialize_type(String.slice(ty, 0, String.length(ty) - 2))}
+
+        String.length(ty) > 0 and String.slice(ty, 0, 1) == String.upcase(String.slice(ty, 0, 1)) ->
```

Comment:
refresh my memory: is first-letter capital _always_ a type in solidity?

Reply:
I can't find any such reference in Eip-712, but then again, it doesn't say anything about that, so you could define a type `MyType(uint256[])` (where that's the identifier, not the type signature!), or `uint256`, right? So I think following convention seems best here, but I'd prefer it if there were a strict definition?